ql/experimental/callablebonds/callablebond.cpp

```diff
@@ -108,6 +109,270 @@ namespace QuantLib {
     }
 
 
+    /* Expose a mechanism to add spread the engine parameters and
+       clean clean up when the object is destroyed
+     */
+    class EngSpreadHelper {
```

Comment:
You can probably fold this into `NPVSpreadHelper`, instead of having two separate classes.